### ISSUE TITLE
Create a startup task to change Vanity URL "site" field to be a host folder field

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#21889]
For all Content types with base type `VANITY_URL` we need to update the `site` field, which is a custom_field that calls a vtl to a `com.dotcms.contenttype.model.field.HostFolderField`

We will then need to change where we look up the vanity url `site` in code.   

There might be specific logic in the content checkin method for vanity urls that will will need to remove.  I know there is validation that is required here that needs to change.

https://github.com/dotCMS/core/blob/92ece95ec3fe6223d9d74ec756734195229ee558/dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java#L267

**Comment by wezell** [comment]
See #21772

**Comment by dsilvam** [comment]
Passed Internal QA. 
* Upgrading from 22.03 properly converted the field "site" of Vanity URL types from custom field to Site field and kept the site field values of the existing contentlet. Also new vanities work as expected. 
* Starting from fresh shows Vanity URL types with site field of type SIte isteand of custom field and existing vanities show proper values. Also new vanities work as expected. 

Tested with docker image `22.04_5c4ae24_SNAPSHOT`

**Comment by bryanboza** [comment]
We need some work here, after run this upgrade task we are crashing the GraphQL stuff,  tested on authoring server and we run this task: 
```
[12/04/22 19:19:02:715 UTC]  INFO startup.StartupTasksExecutor: Running Upgrade Tasks
[12/04/22 19:19:02:715 UTC]  INFO startup.StartupTasksExecutor: Database version: 220215
[12/04/22 19:19:02:797 UTC]  INFO startup.StartupTasksExecutor: Running Upgrade Tasks: Task220330ChangeVanityURLSiteFieldType
[12/04/22 19:19:03:394 UTC]  INFO startup.StartupTasksExecutor: Database upgraded to version: 220330
[12/04/22 19:19:03:415 UTC]  INFO startup.StartupTasksExecutor: Finishing upgrade tasks.
```

After this we are getting problems trying to generate the graphQL schema. This error has been reported here: #22013

**Comment by freddyucv** [comment]
just cache, it is working now

https://github.com/dotCMS/core/issues/22013#issuecomment-1097174732

**Comment by bryanboza** [comment]
We need to retest the upgrade and also starting with the last version to make sure is not affecting the graphQL stuff

**Comment by dsilvam** [comment]
GraphQL working again when upgrading. Tested on docker from 22.03 to 22.04_23de21d_SNAPSHOT

**Comment by dsilvam** [comment]
Passed Internal QA. Tested the following scenarios:

* Upgrading from 22.03 to 22.04 as mentioned above.
* Starting fresh from full starter `20220414`.
* Starting fresh from empty starter `empty_20220407`.

For each scenario checked that:
* The field type of the "site" field of new and existing Vanity URL types was _Site or Folder_.
* Creating new content from those types.
* Fetching those contents from GraphQL including fetching info of its _site_ field.

**Comment by bryanboza** [comment]
Fixed, tested on release-22.04 // Docker // FF
